protocol/synthetix/contracts/modules/core/VaultModule.sol

```diff
@@ -252,6 +365,277 @@ contract VaultModule is IVaultModule {
+            }
+        }
+    }
+
+    /**
+     * @inheritdoc IVaultModule
+     */
+    function getNetDelegatedPerCollateral(
+        uint128 accountId,
+        address collateralType
+    ) external view override returns (int256) {
+        return
+            AccountDelegationIntents.loadValid(accountId).netDelegatedAmountPerCollateral[
+                collateralType
+            ];
+    }
+
+    /**
+     * @inheritdoc IVaultModule
+     */
+    function getExecutableDelegationAccumulated(
+        uint128 accountId,
+        uint128 poolId,
+        address collateralType
+    ) external view override returns (int256 accumulatedIntentDelta) {
+        uint256[] memory intentIds = AccountDelegationIntents.loadValid(accountId).intentIdsByPair(
+            poolId,
+            collateralType
+        );
+        accumulatedIntentDelta = 0;
+        for (uint256 i = 0; i < intentIds.length; i++) {
+            DelegationIntent.Data storage intent = DelegationIntent.load(intentIds[i]);
+            if (!intent.intentExpired()) {
+                accumulatedIntentDelta += intent.deltaCollateralAmountD18;
+            }
+        }
+    }
+
+    /**
+     * @inheritdoc IVaultModule
+     */
+    function requiredDebtRepaymentForUndelegation(
+        uint128 accountId,
+        uint128 poolId,
+        address collateralType,
+        int256 deltaCollateralAmountD18,
+        uint256 collateralPrice
+    ) external view override returns (uint256) {
+        // Identify the vault that corresponds to this collateral type and pool id.
+        if (deltaCollateralAmountD18 > 0) {
+            return 0;
+        }
+
+        Vault.Data storage vault = Pool.loadExisting(poolId).vaults[collateralType];
+
+        int256 debt = vault.currentEpoch().consolidatedDebtAmountsD18[accountId];
+        uint256 effectiveDebt = debt < 0 ? 0 : debt.toUint();
+        if (effectiveDebt == 0) {
+            return 0;
+        }
+
+        uint256 currentCollateralAmount = vault.currentAccountCollateral(accountId);
+
+        if (currentCollateralAmount < (-1 * deltaCollateralAmountD18).toUint()) {
+            revert ExceedingUndelegateAmount(
+                deltaCollateralAmountD18,
+                0,
+                deltaCollateralAmountD18,
+                currentCollateralAmount
+            );
+        }
+
+        uint256 newCollateralAmountD18 = (currentCollateralAmount.toInt() +
+            deltaCollateralAmountD18).toUint();
+
+        uint256 minIssuanceRatioD18 = Pool
+            .loadExisting(poolId)
+            .collateralConfigurations[collateralType]
+            .issuanceRatioD18;
+
+        uint256 effectiveIssuanceRatioD18 = CollateralConfiguration
+            .load(collateralType)
+            .getEffectiveIssuanceRatio(minIssuanceRatioD18);
+
+        uint256 collateralValue = newCollateralAmountD18.mulDecimal(collateralPrice);
+
+        uint256 maxDebt = effectiveIssuanceRatioD18.mulDecimal(collateralValue);
```

Comment:
I would just try and see if we can tackle the edge case of infinite issuance ratio, but capping the debt to be repaid to the current debt.

Reply:
fixed in last commit here: https://github.com/Synthetixio/synthetix-v3/blob/05cd81fab29ae65d8711ec2a88e9ffa8f58317a1/protocol/synthetix/contracts/modules/core/VaultModule.sol#L520-L522